src/graph/heterograph.h

```diff
@@ -7,11 +7,13 @@
 #ifndef DGL_GRAPH_HETEROGRAPH_H_
 #define DGL_GRAPH_HETEROGRAPH_H_
 
+#include <dgl/runtime/shared_mem.h>
 #include <dgl/base_heterograph.h>
 #include <dgl/lazy.h>
 #include <utility>
 #include <string>
 #include <vector>
+#include <set>
```

Comment:
Where's set used?

Reply:
Used in the parameter `fmts` of function `CopyToSharedMem`. Maybe we can use `dgl_format_code_t` and bit operation instead?

Reply:
But it's only needed in cc file right? I'm fine with using set

Reply:
Also usually people use `unordered_set` more often, but here it's not critical which I think is fine

Reply:
It is needed in the  of the function declaration